### PR TITLE
Combine multiple `--pattern` arguments

### LINKED ISSUE #2000: Specifying multiple pattern arguments the last one hides the others
I've only recently discovered that when you specify multiple include patterns, they are not automatically joined, but *ignored silently* by rspec.

Consider the following spec file:
```
--pattern features/*_spec.rb
--pattern features/plugins/*_spec.rb
```

My assumption was that this behaves identical to ```--pattern features/*_spec.rb,features/plugins/*_spec.rb```.
Instead, it silently ingnores the ```features/*_spec.rb``` files. This is contra-intuitive to how most other tools parse file-level arguments.

Rspec should either bail out when it ignores an include pattern it would ignore, or (preferably) join the patterns.

### PR DESCRIPTION
**Pull request by JonRowe**
Fixes #2000

### CODE REVIEW

File: spec/rspec/core/option_parser_spec.rb [hunk 1]
```diff
@@ -214,6 +214,10 @@ def generate_help_text
           options = Parser.parse([option, 'spec/**/*.spec'])
           expect(options[:pattern]).to eq('spec/**/*.spec')
         end
+        it 'will combine multple patterns' do
```
Comment (myronmarston) [inline spec/rspec/core/option_parser_spec.rb]: A blank line between examples would be nice for consistent formatting :).

Also, the `will` seems superfluous...why not `it 'combines multiple patterns'`?

**Comment by myronmarston** [comment]
Have you tried this from the CLI to confirm it works correctly?

My knowledge of glob patterns is a little week so it LGTM to be but I'd have to test it manually to confirm it works properly.

**Comment by JonRowe** [comment]
Yep I checked :)

**Comment by myronmarston** [comment]
LGTM then :).

**Comment by JonRowe** [comment]
I had to disable cyclomatic complexity for the option parser, I don't think it's worth refactoring as isn't that complex, it's just the option parser DSL

**Pull request merged**